update Lighthouse a11y docs (#4337)
GoogleChrome/web.dev: src/site/content/en/lighthouse-accessibility/aria-name/index.md

set of [ARIA roles](https://www.w3.org/TR/wai-aria-practices-1.1/#aria_ex). Elements
that have any of the following ARIA roles but don't have accessible names will cause this audit to fail:
- `aria-command-name` - `button`, `link`, `menuitem`
- `aria-input-field-name` - `combobox`, `listbox`, `searchbox`, `slider`, `spinbutton`, `textbox`
- `aria-meter-name` - `meter`
- `aria-progressbar-name` - `progressbar`
- `aria-toggle-field-name` - `checkbox`, `menu`, `menuitemcheckbox`, `menuitemradio`, `radio`, `radiogroup`, `switch`
- `aria-tooltip-name` - `tooltip`
- `aria-treeitem-name` - `treeitem`

{% include 'content/lighthouse-accessibility/use-native.njk' %}

{% include 'content/lighthouse-accessibility/scoring.njk' %}

## Example 1: How to add accessible names to your custom ARIA toggle fields

### Option 1: Add inner text to the element

The easiest way to provide an accessible name for most elements
is to include text content inside the element.

For example, this custom checkbox will be announced as "Newspaper"
to assistive technology users:

```html
<div id="checkbox1" role="checkbox">Newspaper</div>

> ```

Reviewer: Using the "clip pattern" described in this post (https://www.a11yproject.com/posts/2013-01-11-how-to-hide-content/) you can hide the inner text on screen, but still have it announced by assistive technology. This can be especially handy if you translate your pages for localization.

```html
<a href="https://web.dev/accessible">Learn more <span class="visually-hidden">about accessibility on web.dev</span></a>
```

### Option 2: Add an `aria-label` attribute to the element
If you can't add inner text—for example, if you don't want
the element's name to be visible—use
the `aria-label` attribute.

This custom switch will be announced as "Toggle blue light"
to assistive technology users:

```html
<div id="switch1"
    role="switch"
    aria-checked="true"
    aria-label="Toggle blue light">
    <span>off</span>
    <span>on</span>
</div>
```

### Option 3: Refer to another element using `aria-labelledby`

Use the `aria-labelledby` attribute to identify another element, using its ID,
to serve as the name for the current element.

For example, this custom menu radio button refers to the `menuitem1Label` paragraph
as its label and will be announced as "Sans-serif":

```html
<p id="menuitem1Label">Sans-serif</p>
<ul role="menu">
    <li id="menuitem1"
        role="menuitemradio"
        aria-labelledby="menuitem1Label"
        aria-checked="true"></li>
</ul>
```

## Example 2: How to add accessible names to your custom ARIA input fields

The easiest way to provide an accessible name for most elements
is to include text content in the element.

> However, custom input fields typically don't have inner text,
> so you can use one of two strategies instead.

Reviewer: ```suggestion
However, custom input fields typically don't have inner text,
so you can use one of the following strategies instead.
```

Reviewer: Originally I thought this was telling me to look up the page at the previously presented options

### Option 1: Add an `aria-label` attribute to the element

Use the `aria-label` attribute to define the name for the current element.

For example, this custom combobox will be announced as "country"
to assistive technology users:

```html

> <div id="combo1" aria-label="country" role="combobox">England</div>

Reviewer: should the inner text be removed from this example?

Author: I don't know–I just copied this example from the page being deleted. Deleting inner text seems fine tho.

---
layout: post
title: "ARIA items do not have accessible names"
description: |
  Learn how to improve your web page's accessibility by making sure that
  assistive technology users can access the names of ARIA items.
date: 2020-12-03
web_lighthouse:
  - aria-command-name
  - aria-input-field-name
  - aria-meter-name
  - aria-progressbar-name
  - aria-toggle-field-name
  - aria-tooltip-name
  - aria-treeitem-name
---

{% include 'content/lighthouse-accessibility/about-aria.njk' %}

{% include 'content/lighthouse-accessibility/accessible-names.njk' %}

## How Lighthouse identifies ARIA items without accessible names

[Lighthouse](https://developers.google.com/web/tools/lighthouse/)
flags custom ARIA items whose names
aren't accessible to assistive technologies:

<figure class="w-figure">
  <img class="w-screenshot" src="aria-toggle-field-name.png"
    alt="Lighthouse audit showing custom toggle elements without accessible names">
</figure>

There are 7 audits that check for accessible names, each one covers a different
set of [ARIA roles](https://www.w3.org/TR/wai-aria-practices-1.1/#aria_ex). Elements

> that have any of the following ARIA roles but don't have accessible names will cause this audit to fail:
> - `aria-command-name` - `button`, `link`, `menuitem`

Reviewer 2: ```suggestion
that have any of the following ARIA roles but don't have accessible names will cause this audit to fail:

- `aria-command-name` - `button`, `link`, `menuitem`
```
I believe I've seen some markdown renderers fail to see this as a paragraph followed by a list, so I'll add a newline to defend against that

- `aria-input-field-name` - `combobox`, `listbox`, `searchbox`, `slider`, `spinbutton`, `textbox`
- `aria-meter-name` - `meter`
- `aria-progressbar-name` - `progressbar`
- `aria-toggle-field-name` - `checkbox`, `menu`, `menuitemcheckbox`, `menuitemradio`, `radio`, `radiogroup`, `switch`
- `aria-tooltip-name` - `tooltip`
- `aria-treeitem-name` - `treeitem`

{% include 'content/lighthouse-accessibility/use-native.njk' %}

{% include 'content/lighthouse-accessibility/scoring.njk' %}

## Example 1: How to add accessible names to your custom ARIA toggle fields

### Option 1: Add inner text to the element

The easiest way to provide an accessible name for most elements
is to include text content inside the element.

For example, this custom checkbox will be announced as "Newspaper"
to assistive technology users:

```html
<div id="checkbox1" role="checkbox">Newspaper</div>
```

> Using the ["clip pattern"](https://www.a11yproject.com/posts/2013-01-11-how-to-hide-content/) you can hide the inner text on screen, but still have it announced by assistive technology. This can be especially handy if you translate your pages for localization.

Reviewer 2: ```suggestion
Using the [clip pattern](https://www.a11yproject.com/posts/2013-01-11-how-to-hide-content/) you can hide the inner text on screen, but still have it announced by assistive technology. This can be especially handy if you translate your pages for localization.
```
